internal-enrichment/shodan/src/shodanImport.py

```diff
@@ -29,310 +40,466 @@ def __init__(self):
         )
         self.shodanAPI = shodan.Shodan(self.token)
 
-    def _generate_host_description(self, shodanHostResponse):
-        # Generate Hostname Desc Block
-        Hostnames = "Hostnames:"
-        for host in shodanHostResponse["hostnames"]:
-            Hostnames = Hostnames + f"\n  - {host}"
+        # Shodan Identity
+        self.shodan_identity = self.helper.api.identity.create(
+            type="Organization",
+            name=self.helper.get_name(),
+            description=f"Connector Enrichment {self.helper.get_name()}",
+        )
+
+    def _get_entity_in_opencti(self, data):
+        opencti_entity = self.helper.api.stix_cyber_observable.read(
+            id=data["entity_id"]
+        )
+        if opencti_entity is None:
```

Comment:
Pair review:

This part throw an error, suggestions: passe when it is not IpV4